src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1023,7 +1023,7 @@ static bool TryParseRareTypes(RuntimeType rt, ReadOnlySpan<char> value, bool ign
 
                 if (throwOnFailure)
                 {
-                    Number.ThrowOverflowException(Type.GetTypeCode(typeof(TUnderlying)));
+                    ThrowHelper.ThrowOverflowException();
```

Comment:
The underlying here was only hitting the "rare types" (`float`, `double`, `nint`, `nuint`, or `char`). None of these were handled by `Number.ThrowOverflowException(TypeCode)` and would have triggered asserts in debugging (since they aren't decimal)  and using an incorrect string otherwise.

Reply:
Directly throwing `OverflowException` without a string is both simpler and more accurate than what we had before.